code/modules/mob/living/basic/slime/slime.dm

```diff
@@ -0,0 +1,394 @@
+
+	RegisterSignal(src, COMSIG_HOSTILE_PRE_ATTACKINGTARGET, PROC_REF(on_slime_pre_attack))
+	RegisterSignal(src, COMSIG_ATOM_ATTACK_HAND, PROC_REF(on_attack_hand) )
+
+	ai_controller.set_blackboard_key(BB_SLIME_EVOLVE, evolve_action)
+	ai_controller.set_blackboard_key(BB_SLIME_REPRODUCE, reproduce_action)
+
+/mob/living/basic/slime/Destroy()
+
+	QDEL_NULL(evolve_action)
+	QDEL_NULL(reproduce_action)
+
+	return ..()
+
+///Random slime subtype
+/mob/living/basic/slime/random
+
+/mob/living/basic/slime/random/Initialize(mapload, new_colour, new_life_stage)
+	return ..(mapload, pick(subtypesof(/datum/slime_type)), prob(50) ? SLIME_LIFE_STAGE_ADULT : SLIME_LIFE_STAGE_BABY)
+
+///Friendly docile subtype
+/mob/living/basic/slime/pet
+	hunger_disabled = TRUE
+
+/mob/living/basic/slime/pet/Initialize(mapload, new_colour, new_life_stage)
+	. = ..()
+	set_pacified_behaviour()
+
+//Hilbert subtype
+/mob/living/basic/slime/hilbert
+
+/mob/living/basic/slime/hilbert/Initialize(mapload, new_colour, new_life_stage)
+	. = ..(mapload, /datum/slime_type/bluespace)
+	ai_controller?.set_blackboard_key(BB_SLIME_RABID, TRUE)
+
+//Overriden base procs
+
+/mob/living/basic/slime/adjust_nutrition(change, forced)
+	. = ..()
+	nutrition = min(nutrition, SLIME_MAX_NUTRITION)
+
+/mob/living/basic/slime/set_nutrition(set_to, forced = FALSE)
+	. = ..()
+	nutrition = min(nutrition, SLIME_MAX_NUTRITION)
+
+
+/mob/living/basic/slime/update_name()
+	///Checks if the slime has a generic name, in the format of baby/adult slime (123)
+	var/static/regex/slime_name_regex = new("\\w+ (baby|adult) slime \\(\\d+\\)")
+	if(slime_name_regex.Find(name))
+		var/slime_id = rand(1, 1000)
+		name = "[slime_type.colour] [life_stage] slime ([slime_id])"
+		real_name = name
+	return ..()
+
+/mob/living/basic/slime/regenerate_icons()
+	cut_overlays()
+	var/icon_text = "[slime_type.colour] [life_stage] slime"
+	icon_dead = "[icon_text] dead"
+	if(stat != DEAD)
+		icon_state = icon_text
+		if(current_mood && !stat)
+			add_overlay("aslime-[current_mood]")
+	else
+		icon_state = icon_dead
+	..()
+
+/mob/living/basic/slime/get_status_tab_items()
+	. = ..()
+	if(!hunger_disabled)
+		. += "Nutrition: [nutrition]/[SLIME_MAX_NUTRITION]"
+		. += "Growth: [amount_grown]/[SLIME_EVOLUTION_THRESHOLD]"
+		. += "Power Level: [powerlevel]/[SLIME_MAX_POWER]"
+
+/mob/living/basic/slime/MouseDrop(atom/movable/target_atom as mob|obj)
+	if(isliving(target_atom) && target_atom != src && usr == src)
+		var/mob/living/food = target_atom
+		if(can_feed_on(food))
+			start_feeding(food)
+	return ..()
+
+///Slimes can hop off mobs they have latched onto
+/mob/living/basic/slime/resist_buckle()
+	if(isliving(buckled))
+		buckled.unbuckle_mob(src,force=TRUE)
+
+//slimes can not pull
+/mob/living/basic/slime/start_pulling(atom/movable/moveable_atom, state, force = move_force, supress_message = FALSE)
+	return
+
+/mob/living/basic/slime/get_mob_buckling_height(mob/seat)
+	if(..())
+		return 3
```

Comment:
```suggestion
/mob/living/basic/slime/get_mob_buckling_height(mob/seat)
	if(..() != 0)
		return 3
```

i think we should rethink this proc as a whole but let's treat the integer like an integer